speech/check_dataVals.m

```diff
@@ -157,32 +180,73 @@ function reload_dataVals(src,evt)
+        %trials that are not just nans
+        maxDiffs = zeros(1,UserData.nSigs);
+        bNaNVals = all(isnan(dataVals(i).(UserData.sigs2plot{1})));
+        if strcmp(UserData.sigs2plot{s},'f0') %for f0 exclude first 40 ms of signal
+            %find trial with >1 sample
+            for j = 1:length(UserData.dataVals)
+                if length(UserData.dataVals(j).pitch_taxis) > 1
+                    tstep = diff(UserData.dataVals(j).pitch_taxis(1:2)); % get time step for each sample
+                    break;
+                end
+            end
+            onset = round(.04/tstep);
+        else
+            onset = 2; %exclude first sample from formant tracks when checking for NaNs
+        end
+        if ~dataVals(i).bExcl && ~bNaNVals
+            for s = 1:UserData.nSigs
+                if strcmp(UserData.sigs2plot{s},'f0') %for f0 exclude first 40 ms of signal
+                    onset = round(.04/tstep);
+                    if length(dataVals(i).(UserData.sigs2plot{s}))>onset+1
+                        maxDiffs(s) = max(abs(diff(dataVals(i).(UserData.sigs2plot{s})(onset:end))));
+                    elseif length(dataVals(i).(UserData.sigs2plot{s})) == 1 %can't diff only one sample
+                        maxDiffs(s) = NaN;
+                    else
+                        maxDiffs(s) = max(abs(diff(dataVals(i).(UserData.sigs2plot{s}))));
+                    end
+                else % for formants use whole track
+                    if length(dataVals(i).(UserData.sigs2plot{s})) == 1 %can't diff only one sample
+                        maxDiffs(s) = NaN;
+                    else
+                        maxDiffs(s) = max(abs(diff(dataVals(i).(UserData.sigs2plot{s}))));
+                    end
+                end
+            end
+        end
+        
         if dataVals(i).bExcl
             badTrials = [badTrials dataVals(i).token]; %#ok<*AGROW>
+        elseif dataVals(i).ampl_taxis(1) < .01
+            earlyTrials = [earlyTrials dataVals(i).token];
         elseif dataVals(i).dur < UserData.errorParams.shortThresh %check for too short trials
             shortTrials = [shortTrials dataVals(i).token];
         elseif dataVals(i).dur > UserData.errorParams.longThresh %check for too long trials
             longTrials = [longTrials dataVals(i).token];
-        elseif find(isnan(dataVals(i).f1(2:end))) %check if there are NaN values in formant tracks, excepting 1st sample
+        elseif any(maxDiffs > UserData.errorParams.jumpThresh) %check for trials any jump above threshold
+            for s = 1:UserData.nSigs
+                if maxDiffs(s) > UserData.errorParams.jumpThresh
+                    jumpTrials.(UserData.sigs2plot{s}) = [jumpTrials.(UserData.sigs2plot{s}) dataVals(i).token];
+                end
+            end
+        elseif find(isnan(dataVals(i).(UserData.sigs2plot{s})(onset:end))) %check if there are NaN values in first signal, excepting 1st sample for formants, 40ms for f0
```

Comment:
This doesn't appear to be inside a loop in which "s" is defined. From the comment, looks like {s} should be {1}?

Reply:
L244 addressed in b29105ca281f505c83339ce68d5f78c19149b596